From: =?utf-8?q?Zbigniew_J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Tue, 22 Sep 2020 19:05:17 +0200
Subject: test-seccomp: accept ENOSYS from sysctl(2) too

It seems that kernel 5.9 started returning that.

(cherry picked from commit 0af05e485a3a88f454c714901eb6109307dc893e)
---
 src/test/test-seccomp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/test/test-seccomp.c b/src/test/test-seccomp.c
index 051b5a2..3df6072 100644
--- a/src/test/test-seccomp.c
+++ b/src/test/test-seccomp.c
@@ -316,7 +316,7 @@ static void test_protect_sysctl(void) {
         if (pid == 0) {
 #if defined __NR__sysctl && __NR__sysctl >= 0
                 assert_se(syscall(__NR__sysctl, NULL) < 0);
-                assert_se(errno == EFAULT);
+                assert_se(IN_SET(errno, EFAULT, ENOSYS));
 #endif
 
                 assert_se(seccomp_protect_sysctl() >= 0);
